### DISCUSSION TITLE
What do you need?

### DISCUSSION THREAD

**Discussion by corysimmons** [founded-labs/react-native-reusables#55] [General]
Can you create a GH Project or push a TODO.md with busy-work things you need? I'd love for this project to be a success.

**Comment by mrzachnugent** [comment]
Glad to see that you'd like to help.

I created a [project](https://github.com/users/mrzachnugent/projects/1) for writing documentation and created [issues](https://github.com/mrzachnugent/react-native-reusables/issues) for each item.

I'll add more issues and create another project for other types of TODOs.

**Reply by corysimmons** [reply]
I didn't notice the example starter, so I stood this up after I initialized a project with https://github.com/expo/examples/tree/master/with-router-tailwind

Getting this up and running following the docs could have been better (I know the project is brand new, so I understand). Just some thoughts before I forget them:

- The code snippets in the docs have references to `@rnr/xyz-primitive` in it which cause type errors. Those should all be relative imports. They could be absolute imports, but then we have to track what they've set their alias paths to, so relative is probably better.
- This project should have a CLI. I'd happily assemble that if you're not working on one yet. Moreover, we should PR https://github.com/expo/examples that adds RNR to the with-router-tailwind example so people can do `bunx create-expo-app -e router-tailwind-rnr` and then `bunx rn-reuseables add button` to get a fresh project fired up and running with as little friction as possible.

I'm going to go through all these components and check if they mostly work as advertised. This will give me a better feel for the library, then I'll start whipping it in shape. :)

**Comment by mrzachnugent** [comment]
> I didn't notice the example starter, so I stood this up after I initialized a project with https://github.com/expo/examples/tree/master/with-router-tailwind
> 
> Getting this up and running following the docs could have been better (I know the project is brand new, so I understand). Just some thoughts before I forget them:
> 
> * The code snippets in the docs have references to `@rnr/xyz-primitive` in it which cause type errors. Those should all be relative imports. They could be absolute imports, but then we have to track what they've set their alias paths to, so relative is probably better.

I don't see any references to `@rnr/xyz-primitive` in the https://rnr-docs.vercel.app/. If that's what you're talking about, can you point out where it is the case?

Not sure I understand the type errors. The references the `@rnr/xyz-primitive` are due to the monorepo. When copy/pasting, users shouldn't use that import path. For example, the [ui/accordion docs](https://rnr-docs.vercel.app/components/accordion/#installation) use the raw file from [packages/reusables/src/components/ui/accordion](https://github.com/mrzachnugent/react-native-reusables/blob/main/packages/reusables/src/components/ui/accordion.tsx) and it changes the `@rnr/accordion` to `~/components/primitives/accordion`. 

**Ideally, the [docs](https://rnr-docs.vercel.app/) and the `apps/starter-*` apps are the source of  truth for copy/pasting.**

> * This project should have a CLI. I'd happily assemble that if you're not working on one yet. Moreover, we should PR https://github.com/expo/examples that adds RNR to the with-router-tailwind example so people can do `bunx create-expo-app -e router-tailwind-rnr` and then `bunx rn-reuseables add button` to get a fresh project fired up and running with as little friction as possible.
> 

This would be really cool! I would like there to be proper documentation before doing that. I think that making a PR for an expo template can be done before the CLI.

> I'm going to go through all these components and check if they mostly work as advertised. This will give me a better feel for the library, then I'll start whipping it in shape. :)

Let me know how that goes!

**Reply by corysimmons** [reply]
> I don't see any references to @rnr/xyz-primitive in the [rnr-docs.vercel.app](https://rnr-docs.vercel.app/). If that's what you're talking about, can you point out where it is the case?

Ah, sorry, you're exactly right. It not on the site, I was copying directories from the repo and had some `@rnr` imports: https://github.com/search?q=repo%3Amrzachnugent%2Freact-native-reusables%20%40rnr&type=code

**Reply by mrzachnugent** [reply]
I should make that clearer. It was original a single repo which made copy/pasting easier. Converting it to a monorepo made copy/pasting directly from the repo harder.

I added some documentation to use the `starter-base` template https://rnr-docs.vercel.app/getting-started/initial-setup/ 

Do you think it would help if there was a README to each `packages` folder explaining to switch the `@rnr/` with `~/components/primitives/`?

**Reply by corysimmons** [reply]
Hmm it seems like shadcn-ui does the local package import stuff as well, they just get around it by:

1. Hiding their packages in some fairly deeply nested dir.
2. When the docs generate, they swap the path. I think you might already be doing this.
3. Having a CLI so people aren't really tempted to ever copy/paste.
